# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/forms/nsTextControlFrame.cpp
# Commit: 9fa579e2d7b9
# Full Hash: 9fa579e2d7b97c7e6eaf77461d40fe94c06ba8d6
# Author: David Shin <dshin@mozilla.com>
# Date: 2024-11-21 04:47:28
# Description:
#   Bug 1931933: Use CSSMinMax instead of std::clamp in nsTextControlFrame::GetNaturalBaselineBOffset. r=layout-reviewers,emilio
#   
#   There is no guarantee that the block size is positive, since the size can overflow.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D229678
# ==============================================================================

diff -r 5ccabf34d30e -r 9fa579e2d7b9 layout/forms/nsTextControlFrame.cpp
--- a/layout/forms/nsTextControlFrame.cpp	Wed Nov 20 19:49:29 2024 +0000
+++ b/layout/forms/nsTextControlFrame.cpp	Wed Nov 20 19:49:42 2024 +0000
@@ -1243,7 +1243,7 @@
     }
 
     if (aBaselineGroup == BaselineSharingGroup::First) {
-      return Some(std::clamp(mFirstBaseline, 0, BSize(aWM)));
+      return Some(CSSMinMax(mFirstBaseline, 0, BSize(aWM)));
     }
     // This isn't great, but the content of the root NAC isn't guaranteed
     // to be loaded, so the best we can do is the edge of the border-box.
